master/buildbot/process/buildrequestdistributor.py

```diff
@@ -404,13 +406,20 @@ def xform(bldr):
 
         # sort the transformed list synchronously, comparing None to the end of
         # the list
-        def nonecmp(a, b):
+        def nonekey(a):
             if a[0] is None:
-                return 1
-            if b[0] is None:
-                return -1
-            return cmp(a, b)
-        xformed.sort(cmp=nonecmp)
+                b = list(a)
```

Comment:
mmh this implementation looks like much more complicated than it was initially with cmp, and much slower.
I would prefer to find an implementation. 
maybe this:
http://code.activestate.com/recipes/576653-convert-a-cmp-function-to-a-key-function/